repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/samlpolicy/SamlFlow20Test.groovy

```diff
@@ -421,7 +422,35 @@ class SamlFlow20Test extends ReposeValveTest {
                     SC_OK,
                     null,
                     [(CONTENT_TYPE): CONTENT_TYPE_JSON],
-                    '{"nope":{}}')
+                    createMappingJsonWithValues(rules: [[potato: [fries: [yummy: "yes"], hashBrowns: [:]]]]))
+        }
+
+        and: "the Issuer is unique which will force the call to Identity (avoiding the cache)"
+        def samlIssuer = generateUniqueIssuer()
+        def saml = samlResponse(issuer(samlIssuer) >> status() >> assertion(issuer: samlIssuer, fakeSign: true))
+
+        when:
+        def mc = deproxy.makeRequest(
+                url: reposeEndpoint + SAML_AUTH_URL,
+                method: HTTP_POST,
+                headers: [(CONTENT_TYPE): CONTENT_TYPE_FORM_URLENCODED],
+                requestBody: asUrlEncodedForm((PARAM_SAML_RESPONSE): encodeBase64(saml)))
+
+        then: "the client gets back a bad response"
+        mc.receivedResponse.code as Integer == SC_BAD_GATEWAY
+
+        and: "the request doesn't get to the origin service"
+        mc.handlings.isEmpty()
+    }
+
+    def "when Identity returns a malformed mapping policy, Repose should return a 502"() {
```

Comment:
The impression I got from Jorge's e-mail is that (at least for MVP), since Identity is going to be manually entering the mapping policies, it's Identity's fault if the mapping policies are invalid.  My _guess_ is, after users are allowed to set their mapping policies, Identity should be validating them.  If somehow an issue gets past Identity but shows up when we're trying to translate the response in Repose, it still indicates an issue in Identity.